Wangscape/utils.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+#include <type_traits>
+#include <stdexcept>
+
+namespace
+{
+
+template<typename I>
+I _pow(I base, I exp)
```

Comment:
I think, basing on integers ranks, that `_pow(long long, long long)` would be enough.